### PR TITLE
Remove last uses of go-datastructures library

### PR DESCRIPTION
**Pull request by justinsb**
We weren't using it significantly.

### CODE REVIEW

**Comment by cheftako** [comment]
/lgtm
The second directory walk seems a bit unpleasant but this doesn't make it worse.

File: config/tests/samples/create/samples.go [hunk 1]
```diff
@@ -236,65 +235,85 @@ func waitForDeleteToComplete(t *Harness, wg *sync.WaitGroup, u *unstructured.Uns
...
-func LoadSamples(t *testing.T, project testgcp.GCPProject) []Sample {
+// LoadAllSamples loads all the samples.
+func LoadAllSamples(t *testing.T, project testgcp.GCPProject) []Sample {
 	matchEverything := regexp.MustCompile(".*")
-	return loadSamplesOntoUnstructs(t, matchEverything, project)
+	return LoadMatchingSamples(t, matchEverything, project)
 }
 
-func loadSamplesOntoUnstructs(t *testing.T, regex *regexp.Regexp, project testgcp.GCPProject) []Sample {
+// LoadMatchingSamples loads the samples that match the regex
```
Comment (cheftako) [inline config/tests/samples/create/samples.go:244]: Error: vet: config/tests/samples/create/contents_test.go:32:13: undefined: mapSampleNamesToFilePaths

File: config/tests/samples/create/samples.go [hunk 2]
```diff
@@ -236,65 +235,85 @@ func waitForDeleteToComplete(t *Harness, wg *sync.WaitGroup, u *unstructured.Uns
...
-	return samples
+	s := Sample{
+		Name:      sampleKey.Name,
+		Resources: resources,
+	}
+	return s
 }
 
-func mapSampleNamesToFilePaths(t *testing.T, regex *regexp.Regexp) map[string][]string {
+// ListMatchingSamples gets the keys for all samples matching the regex, without loading them.
```
Comment (cheftako) [inline config/tests/samples/create/samples.go:289]: Error: vet: config/tests/samples/create/contents_test.go:32:13: undefined: mapSampleNamesToFilePaths

Comment (justinsb) [reply config/tests/samples/create/samples.go]: Will hopefully be fixed by rebase!

File: pkg/test/resourcefixture/resourcefixture.go [hunk 3]
```diff
@@ -78,50 +78,58 @@ type HeavyFilter func(fixture ResourceFixture) bool
...
 			if fi.IsDir() {
-				q.Put(path.Join(dir, fi.Name()))
 				continue
 			}
 			if !strings.HasSuffix(fi.Name(), ".yaml") {
 				continue
 			}
 			fileNameNoExt := strings.TrimSuffix(fi.Name(), ".yaml")
 			if value, ok := testToFileName[fileNameNoExt]; ok {
-				t.Fatalf("error, conflicting files for test '%v' in '%v': {%v, %v}", fileNameNoExt, dir, value, fi.Name())
```
Comment (cheftako) [inline pkg/test/resourcefixture/resourcefixture.go:105]: Other fatal calls were converted to return errors for this anonymous function. Why retain the fatal call here?

Comment (justinsb) [reply pkg/test/resourcefixture/resourcefixture.go:105]: Done

**Pull request merged**